File: abm-sim-js/src/sim/MarketOrder.js
export default class MarketOrder {
  orderType
  volume
  price
  time
  agentID
  constructor(orderType, volume, price, time, agentID) {
    this.orderType = orderType
    this.volume = volume
    this.price = price
    this.time = time
    this.agentID = agentID
  }

  ToString() {
    return `{orderType: ${this.orderType}, volume: ${this.volume}, price: ${this.price}, time: ${this.time}}`
  }
}

export const OrderType = {
  BUY: 0,
  SELL: 1
}

File: ABM-SIM-DOTNET/src/MarketOrder.cs
namespace ABMSIM {
    public enum OrderTypes {
        BUY, SELL
    }

    public class MarketOrder {
        public readonly OrderTypes orderType;
        public readonly int volume;
        public readonly float price;

        public MarketOrder(OrderTypes orderType, int volume, float price) {
            this.orderType = orderType;
            this.volume = volume;
            this.price = price;
        }
    }
}
